### PR TITLE
Feature/cs-021-1/java parser file reader by text

### PR DESCRIPTION
**Pull request by diego-centella-jalafund**
#### Resolved task/US: ([Implement java parser file reader by text ](https://app.clickup.com/t/868aem2c6))
* * * *

## What's in this pull request?
There was implemented the java parser file reader handled by text one string that has read by one file reader

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
It was tested the file reader input, also the validator for the input string on the file reader java parser

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        ParseResult<CompilationUnit> parseResult = javaParser.parse(text);
+        if (parseResult.isSuccessful() && parseResult.getResult().isPresent()) {
+            CompilationUnit compilationUnit = parseResult.getResult().get();
+            return compilationUnit;
+        } else {
+            throw new RuntimeException("Error to analyze de code");
+        }
+    }
+
+    public void validateInput(String text) {
```
Comment (luiggy-mamani-jalafund) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java]: could you make a class to handle errors or validations and avoid having multiple responsibilities in one class, please?

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class JavaParserFileReaderByText implements IFileAnalyzer<CompilationUnit, String> {
+    @Override
+    public CompilationUnit read(String text) {
+        JavaParser javaParser = new JavaParser(new ParserConfiguration());
+        ParseResult<CompilationUnit> parseResult = javaParser.parse(text);
+        if (parseResult.isSuccessful() && parseResult.getResult().isPresent()) {
+            CompilationUnit compilationUnit = parseResult.getResult().get();
+            return compilationUnit;
+        } else {
+            throw new RuntimeException("Error to analyze de code");
```
Comment (luiggy-mamani-jalafund) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java]: this method could throw an exception, so can you add "throws" to the method to indicate that an exception may be thrown by this method

image

**Review by luiggy-mamani-jalafund — CHANGES_REQUESTED** [review]

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java (hunk 1 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java]: Done

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java (hunk 2 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java]: Done

**Comment by zamoranoalicia** [comment]
No longer valid, a new implementation should be done on this ticket

**Pull request closed without merging**